### PR TITLE
Added health, setHealth, changeDadCharacter, changeBoyfriendCharacter, keyPressed to modcharting

### PR DESCRIPTION
**Pull request by craftersshaft**
messy implementation but just a quick project i whipped up from learning the source code. added most of the documentation to the docs but forgot to mention changeDadCharacter and changeBoyfriendCharacter, but those work by just taking a character name as an argument (for example, "pico" or "senpai".) 

Example of keyPressed:
```lua
function keyPressed(key)
	if key == "left" then
	setActorX(getActorX("boyfriend")-10, "boyfriend")
	end
	if key == "right" then
	setActorX(getActorX("boyfriend")+10, "boyfriend")
	end
	if key == "up" then
	setActorY(getActorY("boyfriend")-10, "boyfriend")
	end
	if key == "down" then
	setActorY(getActorY("boyfriend")+10, "boyfriend")
	end	
end
```

first time doing a pull request so if i need to do anything else let me know

### CODE REVIEW

**Review by isakube — APPROVED** [review]
yo ho ho, pretty good

**Review by puyoxyz — DISMISSED** [review]
`source/PlayState.hx:1836: characters 27-38 : Unknown identifier : luaModchart`

**Comment by craftersshaft** [comment]
whoops, guess i forgot the #if windows part again

File: source/ModchartState.hx [hunk 1]
```diff
@@ -312,6 +328,7 @@ class ModchartState
 				// get some fukin globals up in here bois
 	
 				setVar("difficulty", PlayState.storyDifficulty);
+				setVar("health", PlayState.instance.health);
```
Comment (Kade-github) [inline source/ModchartState.hx]: No point in having this since getProperty(); exists, soon with setProperty(); as well

File: source/PlayState.hx [hunk 2]
```diff
@@ -1833,6 +1833,7 @@ class PlayState extends MusicBeatState
 
 		if (health > 2)
 			health = 2;
+			if (executeModchart && luaModchart != null){luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

File: source/PlayState.hx [hunk 3]
```diff
@@ -2362,6 +2363,7 @@ class PlayState extends MusicBeatState
 						else
 						{
 							health -= 0.075;
+							if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

File: source/PlayState.hx [hunk 4]
```diff
@@ -2557,6 +2559,7 @@ class PlayState extends MusicBeatState
 					combo = 0;
 					misses++;
 					health -= 0.2;
+					if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

File: source/PlayState.hx [hunk 5]
```diff
@@ -2565,6 +2568,7 @@ class PlayState extends MusicBeatState
 					daRating = 'bad';
 					score = 0;
 					health -= 0.06;
+					if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

File: source/PlayState.hx [hunk 6]
```diff
@@ -2576,11 +2580,13 @@ class PlayState extends MusicBeatState
 					goods++;
 					if (health < 2)
 						health += 0.04;
+						if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

File: source/PlayState.hx [hunk 7]
```diff
@@ -2576,11 +2580,13 @@ class PlayState extends MusicBeatState
 					goods++;
 					if (health < 2)
 						health += 0.04;
+						if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
 					if (FlxG.save.data.accuracyMod == 0)
 						totalNotesHit += 0.75;
 				case 'sick':
 					if (health < 2)
 						health += 0.1;
+						if (executeModchart && luaModchart != null){	luaModchart.setVar('health', health);};
```
Comment (Kade-github) [inline source/PlayState.hx]: same with this one

**Review by Kade-github — CHANGES_REQUESTED** [review]
health shouldn't be added

**Comment by Kade-github** [comment]
Looks good now.

**Review by Kade-github — APPROVED** [review]

**Comment by craftersshaft** [comment]
thanks for letting me know, removed health variables but kept the others in for now, also forgot to update the modchart docs to add the extra functions and remove health but they're in now

**Comment by craftersshaft** [comment]
just realized i forgot to remove setHealth, but that'll probably be gone once setProperty exists

**Pull request merged**